src/components/query/visual-query-builder/ClientCacheRow.tsx

```diff
@@ -16,7 +16,7 @@ export const ClientCacheRow = ({ clientCache, onClientCacheChange }: Props) => {
           htmlFor="clientCache"
           tooltip="Enable to cache results in the browser that are older than 15 minutes. Note: Dashboard variable query result will not update when client cache is enabled."
         >
-          <Switch id="clientCache" value={clientCache} onChange={onClientCacheChange} />
+          <Switch value={clientCache} onChange={onClientCacheChange} />
```

Comment:
The `ClientCacheRow` could have a prop that takes the `refId` of the query. You can generate the `id` for the Switch like this

```ts
const id = `clientCache-${refId}`
```

And update the `htmlFor` on the `EditorField` to use that `id` and use it as the `id` for the `Switch` component.

This would still allow clicking on the label to toggle the switch.